README.md

```diff
@@ -0,0 +1,42 @@
+stack-to-nix
+===========
+
+This is a handy nix function for building stack projects with nix, fully inside the nix sandbox.
+It parses the stack.yaml file and translates the dependencies into a series of fetchUrl, fetchGit calls.
+
+## Example
+
+```nix
+{ pkgs }:
+
+let stackToNix = pkgs.callPackage (fetchTarball https://github.com/serokell/stack-to-nix/archive/master.tar.gz) { };
+in
+stackToNix {
+  # root: the path with stack.yaml. you may want to filter this. for example using nix-gitignore.
+  root = ./.;
+  # shell: get the .env instead of the nix-build derivation. recommended that you do this with shell.nix/default.nix.
+  # see https://github.com/DisciplinaOU/disciplina/blob/master/shell.nix
+  shell = pkgs.lib.inNixShell;
```

Comment:
This should not be used because it doesn't stack and ends up altering builds in implicit ways. See https://github.com/NixOS/nixpkgs/issues/51681 for recent example.

Reply:
What do you mean, it doesn't stack?

Reply:
Nix shells stack: you can have a nix-shell inside a nix-shell. `pkgs.lib.inNixShell` can determine that you're in nix-shell, but is not able to determine if it's used as a context for `nix-build` invocation, or used as a target (i.e.`nix-shell default.nix`).

Imagine you're in the following nix-shell: `nix-shell -p ruby`

If you then try to build this Nix snippet with `nix-build`, you will end up with `shell = true;` and no actual build taking place, because you are indeed inside a Nix shell, Nix just doesn't know which one.

Reply:
I see, thank you.